### PR TITLE
feat: SHAP values for binary classification [target: r2025.0]

### PR DESCRIPTION
**Pull request by ahuber21**
# Description
Enable calculation of SHAP values for binary classification models

Changes proposed in this pull request:
- Allocate result similar to GBT regression
- Rely on GBT regression SHAP dispatching, analogous to calculation of probabilities
- Correctness is tested in a matching update in sklearnex, where we compare against SHAP values from XGBoost

ToDos:

- [ ] Potentially add version checks, depending on CI results

### CODE REVIEW

File: cpp/daal/include/algorithms/gradient_boosted_trees/gbt_classification_predict_types.h [hunk 1]
```diff
@@ -68,7 +90,7 @@ enum ResultToComputeId
 };
 
 /**
- * \brief Contains version 2.0 of the Intel(R) oneAPI Data Analytics Library interface.
+ * \brief Contains version 2 .0 of the Intel(R) oneAPI Data Analytics Library interface.
```
Comment (samir-nasibli) [inline cpp/daal/include/algorithms/gradient_boosted_trees/gbt_classification_predict_types.h:93]:
```suggestion
 * \brief Contains version 2.0 of the Intel(R) oneAPI Data Analytics Library interface.
```

**Comment by ahuber21** [comment]
There is now way around the API-breaking change. I will finish the work as much as possible, but merging needs to be postponed to a major release.

**Review by icfaust — APPROVED** [review]
LGTM

**Pull request merged**